### PR TITLE
Creating Gelu op (no backward).

### PR DESCRIPTION
**Pull request by Narsil**

### CODE REVIEW

File: src/tensor.rs [hunk 1]
```diff
@@ -583,6 +584,9 @@ impl Tensor {
                         let sum_grad = grads.or_insert(arg)?;
                         *sum_grad = sum_grad.add(&arg_grad)?
                     }
+                    Op::Gelu(_) => {
+                        todo!("Implement relu bwd")
```
Comment (LaurentMazare) [inline src/tensor.rs:588]: relu?

File: src/op.rs [hunk 2]
```diff
@@ -130,3 +132,29 @@ impl UnaryOp for Sqrt {
     const KERNEL_F32: &'static str = "usqrt_f32";
     const KERNEL_F64: &'static str = "usqrt_f64";
 }
+
+/// `gelu` operation
+/// <https://en.wikipedia.org/wiki/Activation_function#Comparison_of_activation_functions>
+#[inline]
+pub fn gelu_f32(v: f32) -> f32 {
+    0.5 * (v)
```
Comment (LaurentMazare) [inline src/op.rs:140]: Why the additional parenthesis around `(v)`?

File: src/op.rs [hunk 3]
```diff
@@ -130,3 +132,29 @@ impl UnaryOp for Sqrt {
...
+        * (1.0 + f32::tanh((2.0f32 / std::f32::consts::PI).sqrt() * v * (1.0 + 0.044715 * v * v)))
+}
+/// `gelu` operation
+/// <https://en.wikipedia.org/wiki/Activation_function#Comparison_of_activation_functions>
+#[inline]
+pub fn gelu_f64(v: f64) -> f64 {
+    0.5 * (v)
```
Comment (LaurentMazare) [inline src/op.rs:147]: Same as above for `(v)`.

File: kernels/src/unary.cu [hunk 4]
```diff
@@ -17,10 +17,21 @@ extern "C" __global__ void FN_NAME( \
...
+    T alpha = x + fastCoeff * x_cube;
+    return 0.5 * x * (1.0 + tanhg(M_2_SQRTPI * M_SQRT1_2 * alpha));
+}
+
+
 #if __CUDA_ARCH__ >= 530
 UNARY_OP(__half, uneg_f16, -x)
 UNARY_OP(__half, usqr_f16, x*x)
 UNARY_OP(__half, usqrt_f16, sqrtg(x))
+// UNARY_OP(__half, gelu_f16, gelu_fwd(x))
```
Comment (LaurentMazare) [inline kernels/src/unary.cu:34]: Could we uncomment this so that it's actually compiled even if not used?

File: kernels/src/unary.cu [hunk 5]
```diff
@@ -29,3 +40,4 @@ UNARY_OP(float, usqr_f32, x*x)
 UNARY_OP(float, usqr_f64, x*x)
 UNARY_OP(float, usqrt_f32, sqrtg(x))
 UNARY_OP(float, usqrt_f64, sqrtg(x))
+UNARY_OP(float, gelu_f32, gelu_fwd(x))
```
Comment (LaurentMazare) [inline kernels/src/unary.cu:43]: Add `gelu_f64` too?

File: src/tensor.rs (hunk 1 above)
Comment (Narsil) [reply src/tensor.rs:588]: Fixed.

**Comment by Narsil** [comment]
Should have address all comments.

**Review by LaurentMazare — APPROVED** [review]

**Pull request merged**